### DISCUSSION TITLE
v0.6.3 Released

### DISCUSSION THREAD

**Discussion by fvanroie** [HASwitchPlate/openHASP#338] [Releases]
### Commands
- Additional `idle` parameters now accept `off`, `short` and `long`
- Add `sensors` command to trigger the sensors state message
- Run `L:/boot.cmd` when the plate is (re)booted if the script exists
- You can now use the `pXbY.jsonl` command to update multiple object properties at once (thanks @nagyrobi)

### Objects
- All objects have a custom `tag` property which can contain arbitrary JSON data *(or numbers or text)* (thanks @nagyrobi)
- `img.src` now accepts both `http` and `https` urls (thanks @htvekov)
- `img.src` now accepts `png` and `binary` image urls, PSram is *highly* recommended
- `img.src` now accepts 16-bit BMP files stored in flash

### Web UI
- Updated to modern responsive design *(requires JavaScript)*
- Add `/api/info/` and `/api/config/*/` endpoints
- Allow for a customizible `vars.css`, `style.css`, `script.js` and `edit.htm`
- Display a message when the configuration is changed and a reboot is needed
- Add checkbox for backlight inversion to Display settings (thanks @wolffman122)
- Add #261 (thanks @geiseri)
- Allow firmware upgrade/downgrade in AP mode

### GUI
- `antiburn` displays random pixels *aka.* white noise
- Hide cursor during `antiburn` and `idle` if the pointer is enabled

### MQTT
-  Remember last `page` id of `jsonl` messages. Sending multiple messages now behaves like `jsonl` files (thanks @arovak)

### Fonts
- Use TrueType fonts from flash, PSram is *highly* recommended (thanks to @s-hadinger)
- Custom LVGL binary fonts can be read from flash and loaded into memory, PSram is *highly* recommended
- Font ID is replaced by `namexx` *(where `xx` is the font size)* but the previous Font IDs still work for backwards compatibility
- Added build option for Vietnamese character set (thanks @kydang789)
- ⚠️ **Breaking:** Removed defunct .zi font support!

### Bug fixes
- Fixed #316 (thanks @DJBenson)
- Fix for screenshots not showing properly in Safari on macOS/iOS (thanks @masto)
- Fix bug that would not accept `on` state for setting output GPIOs #275 (thanks @freshnas and @cerietke)
- Fix  #299 (thanks @htvekov)
- Fix `src` bug in `img` objects that could corrupt images sent over http (thanks @htvekov)
- Fix screen dimensions in `statusupdate` message, taking into account current orientation #278 (thanks @kquinsland)
- Fix for HTTP password that could be overwritten by 8 asterisks when it was not changed in the web UI
- Fixed #304 (thanks @fake-name)
- Fixed #300 (thank @nagyrobi)
- Fixed a bug that prevented Wifi from connecting to different BSSID #330 (thanks @Braehead)
- Syslog message format fixes and improvements #285 (thanks @geiseri)

### Custom component
- Expose `antiburn` and `page` in the CC (thanks @dgomes)
- Expose the device URL in discovery message and CC

### Architecture
- Moved to Tasmota Arduino 2.0.3 with native LittleFS library (thanks @Jason2866) and ESP-IDF 4.4.1 fixes for FragAttacks CVEs (thanks @nagyrobi)
- Add support for ESP32-S2 devices
- ⚠️ **Breaking:** Removed support for ESP8266!

Updated libraries to ArduinoJson 6.19.4, ArduinoStreamUtils 1.6.3, AceButton 1.9.2, TFT_eSPI 2.4.61, LovyanGFX 0.4.17 and Adafruit STMPE610 1.1.4

**Comment by jazzmonger** [comment]
I see you added support for esp32-S2 chips.  I have a couple of Esp32-s2 mini.  How do I configure for that? I don't see any S2 board options...

EDIT: Ok, I see it now in the Dev branch.  Where are the input/output pins/gpios defined?  They are completely different from the std builds... I do see this
![E438CB45-F85B-45A6-A15C-3E844B187745](https://user-images.githubusercontent.com/52110065/216312339-63e39bcf-d491-4572-acd0-94cd2e9ca6cf.jpeg)
I assume those are the PIN numbers?  There is no correlation I could find in any docs online of  PIN numbers to D2,D2,D3, etc for the ESP32-S2 Mini chip.

What is HSPI?

I have a pin labeled LED on my display, I assume that's wired to pin 36?  Is there a reference somewhere of "D" nomenclature to PIN numbers?

    -D TFT_BCKL=36  ;None, configurable via web UI (e.g. 2 for D4)

lastly, what is pin "-1" ? No connection?

**Reply by fvanroie** [reply]
Unless you have a compelling reason to use the S2 mini, I would advice to use a regular ESP32 or even an ESP32-S3.
The S2 is under specced with only 320kB RAM vs 520 kB in the other series.

**Comment by rarenu** [comment]
Dear sir l request you to release a HASwitchPlate version to support TT Lillygo T-Watch display so that the Home Automation comes to ground breaking Wearable level. TT Lillygo T-Watch has all the hardware features you need to convert it into a HASwitchPlate. Please consider this and come out with a good news soon.

**Comment by apline87** [comment]
Why this build is not available for Sunton ESP32-4827S043 using web installer? It seems very difficult to set up in PlatformIO environment. Or is there any guideline for PlatformIO setup for newbies?
Thank you in advance

**Reply by fvanroie** [reply]
ESP32-S3 boards are only supported since 0.7.0, which isn't formally released yet. You can get the latest dev build from nightly.openhasp.com or the Actions tab in Github.

**Reply by apline87** [reply]
Hi, thank you for fast reply. I installed the last build 0.7.0. I have screen flickering effect. The information on the screen is visible. HTTP works. But Screen can't be discovered in Home Assistant. May I provide some contribution for testing this device?
